cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -689,6 +697,56 @@ func (s *Supervisor) startOpAMPClient() error {
 	return nil
 }
 
+func (s *Supervisor) startHealthCheckServer() error {
+	if s.config.HealthCheck.Port == 0 {
+		return nil
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/health", func(w http.ResponseWriter, _ *http.Request) {
+		if s.persistentState == nil {
+			w.WriteHeader(http.StatusPreconditionFailed)
+			_, _ = w.Write([]byte("persistent state is nil"))
+			return
+		}
+
+		cfg, ok := s.cfgState.Load().(*configState)
+		if !ok {
+			w.WriteHeader(http.StatusPreconditionFailed)
+			_, _ = w.Write([]byte("config state is nil"))
+			return
+		}
+		if cfg == nil {
+			w.WriteHeader(http.StatusPreconditionFailed)
+			_, _ = w.Write([]byte("config state is nil"))
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+	})
+
+	healthCheckServerPort := s.config.HealthCheck.Port
+	s.healthCheckServer = &http.Server{
+		Addr:        fmt.Sprintf("localhost:%d", healthCheckServerPort),
```

Comment:
I'm sorry. That's better, but you want this configurable too - because in docker or ECS type envs, you'd like to bind to 0.0.0.0 or to a specific network interface.

Reply:
No worries, I can make the host interface configurable and default to `localhost`. Should be a small change and I can take some time to rebase with `main` because I'm almost 100 commits behind.